app/lib/history/models.dart

```diff
@@ -36,16 +37,23 @@ class History extends db.ExpandoModel {
   }
 
   factory History.entry(HistoryEvent event) {
+    final pkgEvent = event is PackageHistoryEvent ? event : null;
+    final publisherEvent = event is PublisherHistoryEvent ? event : null;
     return History._(
-      packageName: event.packageName,
-      packageVersion: event.packageVersion ?? '*',
+      publisherId: publisherEvent?.publisherId,
+      packageName: pkgEvent?.packageName,
+      packageVersion:
+          pkgEvent == null ? null : (pkgEvent.packageVersion ?? '*'),
       timestamp: event.timestamp,
       source: event.source,
       event: event,
     );
   }
 
-  @db.StringProperty(required: true)
+  @db.StringProperty()
+  String publisherId;
+
+  @db.StringProperty()
   String packageName;
```

Comment:
Doc string that says this is `null` if it's not package related (ie. publisher related). Maybe the same for `publisherId`.